java/src/main/java/com/google/appengine/tools/pipeline/Job.java

```diff
@@ -385,6 +386,15 @@ private void setCurrentRunGuid(String guid) {
     updateSpec.getNonTransactionalGroup().includeSlot(promisedValue.getSlot());
     return promisedValue;
   }
+  
+  public <F> PromisedValue<F> promise(String promiseHandle) {
```

Comment:
javadoc.
Also, I wonder what is your use case? PromisedValue only expose its handle, so the only value that I can think
of is if you want to "pass around" the Value to deeply nested children and make their run depends on it.
I am pretty sure this is not going to work correctly if you try to use it with different root jobs (so maybe a more elegant form of passing [not by handle] that would only work between parent to child would be preferred). 
In any case, I think you should provide tests for it.